docs/src/dplyr.md

```diff
@@ -0,0 +1,293 @@
+
+Like dplyr, the DataFramesMeta.jl package contains a set of macros (or "verbs") that perform common data manipulation operations such as filtering for rows, selecting specific columns, re-ordering rows, adding new columns and summarizing data. 
+
+In addition, DataFramesMeta.jl contains a useful operation `@combine` to perform another common task which is the "split-apply-combine" concept. We will discuss that in a little bit. 
+
+## How Does It Compare To Using Base Functions in Julia and in DataFrames.jl?
+
+If you are familiar with Julia, you are probably familiar with base Julia functions such  `map`, and `broadcast` (akin to `*apply` in R). These functions are convenient to use, but are designed to work with `Vector`s, not tabular data.
+
+DataFrames provides the functions `select`, `transform`, and more to work with data frames. Unlike `map` and `broadcast`, these functions are designed to work with tabular data, but have a complicated syntax. 
+
+DataFramesMeta.jl provides a convenient syntax for working with the vectors in a `DataFrame` so that you get all the convenience of Base Julia and DataFrames combined. 
+
+## How Do I Get DataFramesMeta.jl? 
+
+To install DataFramesMeta.jl:
+
+```julia
+import Pkg
+Pkg.activate(; temp=true) # activate a temprary environment for this tutorial
+Pkg.add("DataFramesMeta");
+```
+
+To load DataFramesMeta.jl
+
+```@example 1
+using DataFramesMeta
+```
+
+For this tutorial, we will install some additional packages as well. 
+
+```julia dplyr
+Pkg.add(["CSV", "HTTP"])
+```
+
+Now we load them. We also load the Statistics standard library, which is shipped with Julia, so does not need to be installed.
+
+```@example 1
+using CSV, HTTP, Statistics
+```
+
+We will use [CSV.jl](https://csv.juliadata.org/stable/) and [HTTP.jl](https://juliaweb.github.io/HTTP.jl/stable/) for downloading our dataset from the internet.
+
+# Data: Mammals Sleep
+
+The `msleep` (mammals sleep) data set contains the sleep times and weights for a set of mammals and is available in the dagdata repository on github. This data set contains 83 rows and 11 variables.  
+
+We can loads the data directly into a DataFrame from the `url`. 
+
+```@example 1
+url = "https://raw.githubusercontent.com/genomicsclass/dagdata/master/inst/extdata/msleep_ggplot2.csv"
+msleep = CSV.read(HTTP.get(url).body, DataFrame; missingstring="NA")
+```
+
+The columns (in order) correspond to the following: 
+
+column name | Description
+--- | ---
+`:name` | common name
+`:genus` | taxonomic rank
+`:vore` | carnivore, omnivore or herbivore?
+`:order` | taxonomic rank
+`:conservation` | the conservation status of the mammal
+`:sleep_total` | total amount of sleep, in hours
+`:sleep_rem` | rem sleep, in hours
+`:sleep_cycle` | length of sleep cycle, in hours
+`:awake` | amount of time spent awake, in hours
+`:brainwt` | brain weight in kilograms
+`:bodywt` | body weight in kilograms
+
+
+# Important DataFramesMeta.jl Verbs To Remember
+
+dplyr verbs | Description
```

Comment:
why do you call them dplyr verbs?

Reply:
The [base tutorial](https://genomicsclass.github.io/book/pages/dplyr_tutorial.html) this came from uses the term "verb". I think the author likes the term because it sounds less technical than "function".

Reply:
I am OK with `verb`, I am not clear why you use term "dplyr" - it seems these DataFramesMeta.jl verbs.

Reply:
Oh that was a typo, sorry.